core/modules/sql/SqlTransactionStd.cc

```diff
@@ -0,0 +1,82 @@
+/*
+ * LSST Data Management System
+ * Copyright 2015-2016 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "sql/SqlTransactionStd.h"
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+
+namespace {
+LOG_LOGGER _log = LOG_GET("lsst.qserv.util.SqlTransactionStd");
+}
+
+namespace lsst {
+namespace qserv {
+namespace sql {
+
+// Constructors
+SqlTransactionStd::SqlTransactionStd(sql::SqlConnection& conn)
+    : errObj(), trans(conn, errObj) {
+    if (errObj.isSet()) {
+        LOGS(_log, LOG_LVL_ERROR, "Constructor failed (" << errObj.errNo() << ") " << errObj.errMsg());
+        throwException(ERR_LOC, "Constructor failed");
```

Comment:
Calling virtual methods from constructor is asking for trouble ⛔️ Just throw what you want to throw.

Reply:
The things calling this are catching different specific exceptions. Throwing something else will cause them to break. I'll need to think about this.

Reply:
Well, the point is that overriding `throwException` does not work if you call it from constructor, in that respect things are already broken so you need to fix it somehow.

Reply:
More of a need to think about how to fix it as it needs to be fixed.